static/styles/zulip.scss

```diff
@@ -1603,6 +1583,30 @@ div.focused_table {
                 padding: 5px 15px 0px 0px;
             }
         }
+
+        > :not(.search_closed):not(.sub_count):not(.narrow_description) {
+            flex-shrink: 1;
+            overflow: hidden;
+        }
+
+        > :not(.stream) {
+            text-overflow: ellipsis;
+        }
+
+        > :nth-last-child(2) {
+            flex-grow: 1;
+        }
+
+        // We extend the clickable area for the search_closed icon over the
+        // 2nd last navbar element for better design (user convenience) this
+        // selector makes it so mousing over that element also gives the user
+        // a visual cue about the results of clicking the element
+        > :nth-last-child(2) {
+            cursor: pointer;
+            &:hover + .search_closed {
+                color: hsl(0, 0%, 0%);
+            }
+        }
```

Comment:
I don't understand the pseudotag logic here at all; can you explain in the commit message why the new location is correct?  

(Also, I think this whole pseudotag with things like nth-last-child is a code smell)

Reply:
I like the `nth-last-child` trick, I wouldn't want to remove it, maybe just detail the idea as:
```
\\ The very last element in the navbar is the search icon, the second
\\ last element is either the narrow description (for stream narrows) or
\\ the "title" (for other narrows). The flex-grow property ensures these
\\ elements take up the entirety of the white space.
```

hmm...
or maybe we just add a `grow` class and just place it on both of those elements.

Reply:
Yeah, that's fair, as long as it has an explanatory comment it makes sense.